util/src/main/java/org/datacommons/util/StatChecker.java

```diff
@@ -98,12 +97,21 @@ public StatChecker(LogWrapper logCtx, Set<String> samplePlaces, boolean verbose)
     }
   }
 
-  // Given a graph, extract time series info (about the chosen sample places) from the
-  // statVarObservation nodes and save it into seriesSummaryMap.
-  public void extractSeriesInfoFromGraph(McfGraph graph) {
+  // Given a graph, extract stat var info and time series info (about the chosen sample places) from
+  // the statVarObservation nodes.
+  public void extractStatsFromGraph(McfGraph graph) {
     for (Map.Entry<String, McfGraph.PropertyValues> node : graph.getNodesMap().entrySet()) {
-      if (isSvObWithNumberValue(node.getValue()) && shouldExtractSeriesInfo(node.getValue())) {
-        extractSeriesInfoFromNode(node.getValue());
+      if (isSvObWithNumberValue(node.getValue())) {
+        // We will extract basic stat var information from every StatVarObservation nodes
+        extractStatVarInfoFromNode(node.getValue());
```

Comment:
We now do synchronized work for *every* SVObs. I think this will cause more contention. Probably add a TODO to look into optimizing this?

Reply:
done